diff --git a/yt-dlp/yt_dlp/postprocessor/embedthumbnail.py b/yt-dlp/yt_dlp/postprocessor/embedthumbnail.py
index 9ae59a7c3..153064f9d 100644
--- a/yt-dlp/yt_dlp/postprocessor/embedthumbnail.py
+++ b/yt-dlp/yt_dlp/postprocessor/embedthumbnail.py
@@ -140,7 +140,7 @@ def run(self, info):
                 success = True
                 atomicparsley = next((
                     # libatomicparsley.so : See https://github.com/xibr/ytdlp-lazy/issues/1
-                    x for x in ['AtomicParsley', 'atomicparsley', 'libatomicparsley.so']
+                    x for x in ['AtomicParsley', 'atomicparsley', 'libatomicparsley.bin.so']
                     if check_executable(x, ['-v'])), None)
                 if atomicparsley is None:
                     self.to_screen('Neither mutagen nor AtomicParsley was found. Falling back to ffmpeg')
