diff --git a/yt-dlp/yt_dlp/postprocessor/ffmpeg.py b/yt-dlp/yt_dlp/postprocessor/ffmpeg.py
index 1f179c145..a098ff281 100644
--- a/yt-dlp/yt_dlp/postprocessor/ffmpeg.py
+++ b/yt-dlp/yt_dlp/postprocessor/ffmpeg.py
@@ -106,7 +106,7 @@ def _determine_executables(self):

         location = self.get_param('ffmpeg_location', self._ffmpeg_location.get())
         if location is None:
-            return {p: p for p in programs}
+            return {'ffmpeg': 'libffmpeg.bin.so', 'ffprobe': 'libffprobe.bin.so'}

         if not os.path.exists(location):
             self.report_warning(f'ffmpeg-location {location} does not exist! Continuing without ffmpeg')
